gui/src/app/Stanc/Linting.ts

```diff
@@ -0,0 +1,103 @@
+import { StancErrors } from "./Types";
+
+type Position = {
+  startLineNumber: number;
+  startColumn: number;
+  endLineNumber: number;
+  endColumn: number;
+};
+
+// An interface for passing markers (squiggles) to the editor without depending on monaco types
+export type CodeMarker = Position & {
+  message: string;
+  severity: "error" | "warning" | "hint" | "info";
+};
+
+export const stancErrorsToCodeMarkers = (stancErrors: StancErrors) => {
+  const codeMarkers = [
+    ...(stancErrors.errors || []).map((error) =>
+      stancMessageToMarker(error, "error"),
+    ),
+    ...(stancErrors.warnings || []).map((warning) =>
+      stancMessageToMarker(warning, "warning"),
+    ),
+  ];
+
+  return codeMarkers.filter((marker) => marker !== undefined) as CodeMarker[];
```

Comment:
This is fine, no change requested. But I wonder whether it's even possible to generate one of these? Or was this what the linter was complaining about

Reply:
Apologies, that was a bit gnomic. I mean is it possible for this list to even have any undefined elements? Or is the filter just to keep TSC happy

Reply:
Per our conversation, it is possible to generate warnings with no line numbers (e.g. an entirely whitespace stan program) which would cause `undefined`s to be legitimately returned, so this check may be necessary in some edge cases.